### PR TITLE
Adaptive constructor / TextInputAction docs fix

### LINKED ISSUE #35399: SwitchListTile.adaptive has splash effect on iOS
The [`SwitchListTile.adaptive`](https://api.flutter.dev/flutter/material/SwitchListTile/SwitchListTile.adaptive.html) shouldn't have a splash effect (from Material) on iOS devices (eg. when providing the `CupertinoSwitch`), which currently does.

```
Flutter (Channel unknown, v1.7.11, on Mac OS X 10.14.5 18F132, locale pt-PT)
    • Flutter version 1.7.11 at /Users/miguelruivo/DevTools/flutter
    • Framework revision 43b03127d4 (4 days ago), 2019-06-26 23:27:39 -0400
    • Engine revision f3761ba8d0
    • Dart version 2.4.0
```

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
## Description

A few documentation fixes wrt adaptive constructors and `TextInputAction`.

Trying to make it clear in the documentation that adaptive constructors still create Material widgets.

## Related Issues

Fixes: https://github.com/flutter/flutter/issues/35399
Related: https://github.com/flutter/flutter/issues/68571

## Tests

Docs only change.

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I signed the [CLA].
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.
- [ ] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [ ] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/widgets/editable_text.dart [hunk 1]
```diff
@@ -1021,63 +1021,6 @@ class EditableText extends StatefulWidget {
   /// Called when the user indicates that they are done editing the text in the
   /// field.
   /// {@endtemplate}
-  ///
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/widgets/editable_text.dart:1024]: I tried to keep the example by changing it to "moving focus to a specific input field" but found it not as common/practical. @gspencergoog any suggestions?

File: packages/flutter/lib/src/material/switch_list_tile.dart [hunk 2]
```diff
@@ -284,10 +284,15 @@ class SwitchListTile extends StatelessWidget {
...
        super(key: key);
 
-  /// Creates the wrapped switch with [Switch.adaptive].
+  /// Creates a Material [ListTile] with an adaptive [Switch], following
+  /// Material design's
+  /// [Cross-platform guidelines](https://material.io/design/platform-guidance/cross-platform-adaptation.html).
   ///
-  /// Creates a [CupertinoSwitch] if the target platform is iOS, creates a
-  /// material design switch otherwise.
+  /// {@macro flutter.material.switch.adaptive }
```
Comment (gspencergoog) [inline packages/flutter/lib/src/material/switch_list_tile.dart:291]:
```suggestion
  /// {@macro flutter.material.switch.adaptive}
```

File: packages/flutter/lib/src/widgets/editable_text.dart (hunk 1 above)
Comment (gspencergoog) [reply packages/flutter/lib/src/widgets/editable_text.dart:1024]: I think it's fine to remove this if it isn't relevant anymore.

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Review by justinmc — APPROVED** [review]
LGTM 👍

File: packages/flutter/lib/src/material/switch_list_tile.dart [hunk 3]
```diff
@@ -284,10 +284,15 @@ class SwitchListTile extends StatelessWidget {
...
-  /// Creates the wrapped switch with [Switch.adaptive].
+  /// Creates a Material [ListTile] with an adaptive [Switch], following
+  /// Material design's
+  /// [Cross-platform guidelines](https://material.io/design/platform-guidance/cross-platform-adaptation.html).
   ///
-  /// Creates a [CupertinoSwitch] if the target platform is iOS, creates a
-  /// material design switch otherwise.
+  /// {@macro flutter.material.switch.adaptive}
+  ///
+  /// This constructor creates a Material [ListTile] with a platform-dependent
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/switch_list_tile.dart:293]: I think this overstates the truth. We're only "adapting" for iOS, i.e. we're adapting the ListTile's Material switch component to look a little more familiar when it appears on iOS.

File: packages/flutter/lib/src/material/switch.dart [hunk 4]
```diff
@@ -88,8 +88,18 @@ class Switch extends StatefulWidget {
...
         super(key: key);
 
-  /// Creates a [CupertinoSwitch] if the target platform is iOS, creates a
-  /// material design switch otherwise.
+  /// Creates an adaptive [Switch] based on the target platform, following
+  /// Material design's
+  /// [Cross-platform guidelines](https://material.io/design/platform-guidance/cross-platform-adaptation.html).
+  ///
+  /// {@template flutter.material.switch.adaptive}
+  /// The Material design guidelines recommend using the native switches on iOS
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/switch.dart:96]: Although the Material design guidelines do in fact recommend using native iOS switches, we clearly do not do that. The most that we can claim here, is that we'll create switches that resemble the native iOS versions.

**Review by HansMuller — APPROVED** [review]
LGTM with a few additional tweaks

File: packages/flutter/lib/src/material/switch.dart (hunk 4 above)
Comment (LongCatIsLooong) [reply packages/flutter/lib/src/material/switch.dart]: updated.

**Pull request merged**